docs/architecture/platform-versions.md

```diff
@@ -0,0 +1,190 @@
+  tested on it in a long time if ever.  Shortly thereafter, we
+  confirmed that the app rendered quite badly there.
+* We [dropped iOS 8 support][] in 2018-08.  It represented <1% of iOS
+  users who tried Zulip, and we'd learned that it didn't run there.
+* We [dropped iOS 9 support][] in 2019-07.  It was 0.4% of iOS users
+  who tried Zulip, and an Xcode upgrade had dropped iOS 9 from the
+  simulator.
+
+[dropped-android-j]: https://chat.zulip.org/#narrow/stream/48-mobile/topic/platform.20versions/near/625585
+[dropped iOS 8 support]: https://chat.zulip.org/#narrow/stream/48-mobile/topic/platform.20versions/near/628412
+[dropped iOS 9 support]: https://chat.zulip.org/#narrow/stream/48-mobile/topic/platform.20versions/near/771761
+
+
+Related observations:
+
+* Our userbase tends to be far more updated than the ecosystem at
+  large, especially on Android.
+
+  * For example, as of May 2019, Android 5 "Lollipop" (released in
+    2014) and later had reached 88.5% of all devices connecting to
+    Google Play; among users with Zulip installed, it had reached
+    99.1% in figures from the previous month.
+
+  * Broadly, the ratio new:old of devices which have at least a given
+    version vs. those with an earlier version tends to be about 5-10x
+    more for our Android users than for those on Google Play at large.
+
+
+## Browser versions (for the WebView for the message list)
+
+### Policy
+
+For our current minimum versions of Chrome and (Mobile) Safari, see
+the block comment at the top of `js.js`.
+
+General policy:
+
+* Our basic approach is the same as for OS versions:
+
+  * We routinely test and develop on up-to-date platforms.
+
+  * We support older versions primarily by a lazy algorithm; plus
+    before using a fancy feature we consult a compatibility table,
+    e.g. on MDN or on caniuse.com.
+
+  * When an older version (and below) falls below 1-2% of our overall
+    userbase, we simply drop support for it if faced with significant
+    effort to keep it working.
+
+* On iOS, there's little more to say, because the Safari version
+  corresponds directly to the iOS version.  So for example so we'll
+  support Mobile Safari 13 for exactly as long as we support iOS 13.
+
+* On Android, we have additional choices, because the Chrome version
+  (including in a WebView) is generally much more recent than the OS.
+
+  * As a result, we may drop support for old Chrome versions as much
+    as 2-3 years more recent than our oldest supported Android
+    version.
+
+  * Our support thresholds are always a version found in a stock
+    emulator image for some past Android release, for practicality of
+    testing.
+
+
+### Data and commentary
+
+Empirical details on Android Chrome versions found in WebViews:
+
+* Starting in Android 4.4 KitKat (which is <= our minimum supported
+  Android version), the browser in a WebView is updated as an APK 🎉,
+  independently of the OS...
```

Comment:
Hmm! Thanks for fact-checking this.

This text is basically moved straight from the comment `js.js`, and at the time I wrote that I thought it was based on a pretty clear statement from upstream. I'll go try to track down the source for that and see if we can reconcile why they seem to disagree, or what I misread, or what.

Reply:
Well -- the source I [cited then](https://chat.zulip.org/#narrow/stream/48-mobile/topic/platform.20versions/near/627005) is the same page you just linked to. And, double-checking [via the Wayback Machine](http://web.archive.org/web/20180924185906/https://developer.chrome.com/multidevice/webview/overview)... it had the same text then as it does now. :man_shrugging: Seems I read it a bit too hastily, then.

(The text starts with "Since Android 4.4 (KitKat)," but that only applies to being based on Chromium -- which it sounds like was a major one-time upgrade, but didn't yet go with being an independently-upgradable APK.)